### PR TITLE
[WIP]Add support for Object Attributes in AST

### LINKED ISSUE #5338: Allow Object Attributes in AST
## Description
Allow for object attributes to be allowed in the allowlist AST.
The issue being that they do not exist at Class analysis time, and we would want the bound function to have the context of the target data item.

One method could be to provide a flag that these are dynamic and not bind the method but attach generated function to the Pointer so that on the retrieval side we can attempt to locate the attribute at execution time. 

## Definition of Done
Items like the following can be enabled and work as expected:
```
allowlist["torch.nn.Linear.weight"] = "torch.Tensor"
allowlist["torch.nn.Linear.bias"] = "torch.Tensor"
```

### PR DESCRIPTION
**Pull request by avinsit123**
## Description
Closes #5338 

This PR intends to add attributes to AST which are bound to the object not the class. For eg: `torch.nn.Linear(1,2).weight`.
Currently, trying out approach suggested by @madhavajay in #5338. 

### Fix

1. Enable addition of object attributes in AST
2. Add flag to such object attributes in their respective classes
3. Attach Function/Static Attribute to pointer to enable retrieval during execution time

## How has this been tested?
- Will write tests in `tests/syft/lib`

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

**Comment by madhavajay** [comment]
Hi @avinsit123 This issue has been assigned to @tudorcebere because theres a few issues we want to double check with the implementation and how it fits into our existing AST code.

Sorry, about that.

**Comment by avinsit123** [comment]
Oh I see. wish I had known earlier coz I spent quite some time working on this. Anyways, thanks for informing @madhavajay
